### ISSUE TITLE
Menus design pass

### ISSUE THREAD

**Issue by tonyr59h** [wordpress-mobile/WordPress-Android#4156]
Work closely with @mattmiklic to do a final design pass of the `feature/calypso-menus` branch after the feature is fully implemented. Created this issue so we don't continuously bombard @mattmiklic for design reviews on each feature PR, we'll just have one design pass.

BEFORE STARTING the following issues should be completed and merged into the feature branch:
- [x] #3589 (PR #4145 and PR #4146 )
- [X] #3590 (PR #4102)
- [x] #3591
- [x] #3592 (manually merged into `feature/3591-menu-items`)

Things to check:
- [ ] padding/spacing
- [ ] colors
- [ ] fonts
- [ ] icons (menu item types, "grip" icon for drag&drop, others)
- [ ] string text
- [ ] correct usage of UI elements (buttons, spinners, inputs)
- [ ] @mattmiklic has signed off on all of it

Other notes:
- I just noticed that the Menu icon on the main Masterbar screen uses the square bullet style while the icon on the Menus screen uses the round bullet style

**Comment by mattmiklic** [comment]
I'm finally having a chance to review this branch, and have a few suggestions. This is my first time seeing the Menus design on Android, so I apologize in advance for proposing that we shift a few things around, but I think it will give us a design that feels more native in the end.

Screenshot of current branch:
![before](https://cloud.githubusercontent.com/assets/1396552/16702328/4b49e444-4533-11e6-8694-5c8ac0e388ea.png)

Notes:
- Button styles for trash/save are non-standard for Material Design
- Location of Save icon is non-standard
- Type sizes are non-standard
- Text could benefit from aligning to 16dp and 72dp [keylines](https://material.google.com/layout/metrics-keylines.html?authuser=3)
- "USES" text is in all-caps, which denotes a button in Material Design
- Icons are a mix of gridicons and older noticons
- The lower half of the screen doesn't appear for me when building this branch. I don't know if that's to be expected or not, but as a result I wasn't able to review that portion of the design.

Suggested changes:
![after](https://cloud.githubusercontent.com/assets/1396552/16702405/a7da1bd4-4533-11e6-9fd4-326df45e6483.png)

Notes:
- All text sizes use Material Design-standard sizes
- All icons are now Gridicons at standard sizes
- The area and menu pickers are now styled as standard cards, to help distinguish them from the list below.
- The Save button is relocated to the app bar in the standard location*
- Cards use standard (8dp) margins)
- Text & icons are aligned to keylines

* I'm not sure if the text Save button is completely right, the spec seems to indicate that this is usually an icon. In Contacts, for example, the "Save" button is represented by a checkmark icon. Open to suggestions on whether this should be an icon.

I'd like to get thoughts/feedback on this, then once no other changes are needed I can provide all of the type sizes and measurements.

**Comment by mzorz** [comment]
Great feedback @mattmiklic ! I love all of these points and agree to implement them - I just noticed we didn't merge the menu item changes into the `feature/calypso-menus` branch for you to review the complete thing 😞 - that's why you couldn't see the lower half of the screen. I'm sorry for this, I have just merged everything into the calypso-menus branch now so it should be ready if you want to take a look again.

> I'm not sure if the text Save button is completely right, the spec seems to indicate that this is usually an icon. In Contacts, for example, the "Save" button is represented by a checkmark icon. Open to suggestions on whether this should be an icon.

I think the SAVE text button on the upper right corner is clearer in this context (as opposed to a checkmark icon) as the user actually needs to press it to write the changes on stone. (just fyi we still have a dialog if you make changes but forget to hit SAVE and then attempt to leave the screen). Nevertheless, I'm happy to implement the checkmark if you feel this is the way it should be done 👍 

Please note we didn't pay special attention to icons as we knew we needed to get the right ones in the end 😁 - you'll see the horrendous collage when you get to see the menu items stuff 😄 

Thanks again for your feedback @mattmiklic ! And sorry for the extra trouble with this last merge.

cc @tonyr59h

**Comment by mattmiklic** [comment]
I attempted to look at this branch again on Friday, but ran into lots of Gradle build errors. This morning I managed to update my JDK to the new one, and got the app to build in the simulator, but every time I tap the "Menus" list item, the app crashes. For the sake of time, @mzorz could you provide some screenshots of the latest? I think it will be faster for me to review and make some suggestions based on screenshots than to try to keep debugging my problems building the code.

**Comment by mzorz** [comment]
Sure thing Matt! Will provide them today

**Comment by mzorz** [comment]
Here's a bunch of screenshots https://cloudup.com/cyGLc-MrgLC  - sorry for the daly, @mattmiklic ! ping me if you need more 👍
